### PR TITLE
Fixes for crawler and sorter

### LINKED ISSUE #93: Production code should lack any logic related to testing
Fix how https://github.com/freedomofpress/fingerprint-securedrop/pull/57/commits/4fc9d54c0ceeb2be3e432fb7b8e631198e26be43 was implemented, perhaps using a mixin class as elsewhere.

### LINKED ISSUE #94: Revert reading from environment variable for database password
https://github.com/freedomofpress/fingerprint-securedrop/pull/57/commits/ce1e0adb228cca2cff39a81d3a05154596ee50ff introduced a regression in that we're back to reading from environment variables. There's no need to specify the PGPASSFILE environment variable at all because we place the PGPASSFILE in the default location, and it will automatically be found there. Also, use of environment variables tends to be problematic, and removing them was part of the process of getting Travis running.

### PR DESCRIPTION
**Pull request by redshiftzero**
Fix a couple of issues with the sorter and crawler:

1. It looks like the reason the sorter was immediately exiting is that in 0a6b52538d121e58982c68b7042f90b526e8dd53 the delimiter between directories in config.ini was changed to `\n` but the delimiter the sorter was expecting is `,` so it was no longer parsing any directory URLs. 
2. `config.ini` was not being populated with the prod db values since Ansible was not setting up `config.ini`. Made a minor change to read from `~/.pgpass` (since Ansible is setting that up on the dev VM and the VPSes). This also closes #93 and closes #94

### CODE REVIEW

File: fpsd/database.py [hunk 1]
```diff
@@ -22,36 +22,25 @@ class Database:
...
-                content = f.read().decode("utf-8").replace("\n", "").split(":")
-                database_config["pgpass"] = content[-1]
-
-            self.engine = create_engine(
-                'postgresql://{pguser}:{pgpass}@{pghost}:{pgport}/{pgdatabase}'.format(
+            try:
+                database_config = get_db_creds()
+            except OperationalError as exc:
+                panic("fingerprint-securedrop Postgres support relies on use of a "
+                      "PGPASSFILE. Make sure this file and the env var pointing "
```
Comment (psivesely) [inline fpsd/database.py]: You don't need to set the environment variable if it is in the default location, `"${HOME}/.pgpass"`.

**Review by psivesely — DISMISSED** [review]
I'm concerned about the robustness of this solution. It makes for one less configuration section from `config.ini`, however, it relies on the following assumptions:

1. The database configuration we want to use for production will always be first in our PGPASSFILE.
2. `pgport` and `pguser` will be the same for both our production and test databases.

A compromise would be to hardcode all test database values (besides the password) in the `TestDatabase` class, and continue to have the user maintain the `PGPASSFILE` and `config.ini` themselves. One thing that could make it easier is that if a `PGPASSFILE` is created by Ansible (which happens only when there isn't one already to prevent overwrites/password loss), those values could also be populate the `config.ini`. So as long as you don't change the database configuration you passed in via play-level vars (or didn't, and let Ansible create for you via defaults), everything should jut work™ out of the box/provision.

Not everything is nicely Ansible-ized regarding keeping the `PGPASSFILE` and `config.ini` contents in sync because any way I can think of carries some dangers of overwriting credentials.

**Comment by redshiftzero** [comment]
My goal wasn't to remove the configuration section, but to get the provisioning sorted out so we can get the crawlers working again. Will this PR get the crawlers running and dumping data into the right database the way the VPSes are currently being provisioned? If yes let's get this merged so we can continue crawling. Both of the above assumptions are true in the environments we are running these things: vagrant dev VM, Travis, and on the VPSes, no?

**Comment by psivesely** [comment]
Although these assumptions may be correct in the present moment, this requires that we don't make any changes that break those assumptions 

I think there are easier ways to get things re-running by EOD (as that is our expressed goal) that rely on less assumptions. Namely, I still believe we should hard-code the test database credentials in `TestDatabase` and leave management of `config.ini` production database credentials and `PGPASSFILE` to the user after the initial configuration. Should take 5 minutes to write and run a custom tasklist to set the correct production values in the `config.ini` files of the VPSs. Should not need to be modified again. Making the permanent changes to our playbook ensure that these values are set correctly on first provision for Travis, Vagrant, and if/when we spin up new instances.

What do you think @conorsch?

**Comment by conorsch** [comment]
Applied this feature branch to the prod crawlers. @redshiftzero Can you confirm working sorting and crawling? If so, I vote we merge. 

> Namely, I still believe we should hard-code the test database credentials in `TestDatabase` and leave management of `config.ini` production database credentials and `PGPASSFILE` to the user after the initial configuration.

Open a separate issue for that, since this PR includes the test regression of snipping out the db tests to keep the merge-wheels turnin'. 

>  Should take 5 minutes to write and run a custom tasklist to set the correct production values in the `config.ini` files of the VPSs.

Sounds to me like the `config.ini` file should be converted to full template, so we can set sane defaults (e.g. for use in Travis), and override via vars in any other environment (i.e. prod).

**Comment by redshiftzero** [comment]
Sorter is working on the VM and on the VPSes (just finished another sort), database connections are working now everywhere, but I'm running into tor issues on crawling on the VPSes. I say :+1: to merge this since nothing in this PR is changing anything in the crawler to do with Tor (i.e. these issues are in master)

**Comment by conorsch** [comment]
Thanks for confirming functionality, @redshiftzero! Going to merge as-is.

> I'm running into tor issues on crawling on the VPSes.

If you have a traceback, toss into a separate issue!

**Review by conorsch — APPROVED** [review]
:shipit:

**Comment by redshiftzero** [comment]
yep will construct a coherent description of what is going on and make an issue, thanks @conorsch

**Pull request merged**